File: c/1066.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int pos, neg, pairs, odds;

int pair_or_odd(int num){
    if(num % 2 == 0){
        pairs += 1;
    }else{
        odds += 1;
    }
    return 0;
}

int pos_or_neg(int num){
    if(num > 0){
        pos += 1;
    }else if(num < 0){
        neg += 1;
    }else{
        return 0;
    }
    return 0;
}

int main() {
    pos = 0;
    neg = 0;
    pairs = 0;
    odds = 0;

  int number[5];
  for(int i=0; i<5; i++){
    scanf("%d", &number[i]);
    pair_or_odd(number[i]);
    pos_or_neg(number[i]);
  }
  
  printf("%d valor(es) par(es)\n", pairs);
  printf("%d valor(es) impar(es)\n", odds);
  printf("%d valor(es) positivo(s)\n", pos);
  printf("%d valor(es) negativo(s)\n", neg);
  return 0;
}

File: python/1066.py
def pair_or_odd(num):
    if num % 2 == 0:
        return 1
    else:
        return 0

def pos_or_neg(num):
    if num > 0:
        return 1
    elif(num < 0):
        return 0
    else:
        return -1

def main():
    pairs = 0
    odds = 0
    pos = 0
    neg = 0
    
    for i in range(5):
        num = int(input())
        if(pair_or_odd(num)):
            pairs += 1
        else:
            odds += 1

        if pos_or_neg(num) == 1:
            pos += 1
        elif pos_or_neg(num) == 0:
            neg += 1
  
    print(f"{pairs} valor(es) par(es)"); 
    print(f"{odds} valor(es) impar(es)");
    print(f"{pos} valor(es) positivo(s)");
    print(f"{neg} valor(es) negativo(s)");
  
main()
